lms/djangoapps/api_manager/sessions/views.py

```diff
@@ -105,21 +111,29 @@ def post(self, request):
                     # violate our RESTfulness
                     #
                     engine = import_module(settings.SESSION_ENGINE)
-                    new_session = engine.SessionStore()
-                    new_session.create()
+                    if session_id is None:
+                        session = engine.SessionStore()
+                        session.create()
+                        success_status = status.HTTP_201_CREATED
+                    else:
+                        session = engine.SessionStore(session_id)
+                        success_status = status.HTTP_200_OK
+                        if SESSION_KEY in session:
+                            # Cannot re-login a user who is already logged in:
+                            return Response({}, status=status.HTTP_403_FORBIDDEN)
```

Comment:
Why are we returning 403 if the user is already logged in?  Wouldn't we just return a 200?